### PR TITLE
fix(sheet): freeze bugfix

### LINKED ISSUE #890: [Bug] 冻结行列问题
### 初始清单

- [X] 这真的是个问题吗？
- [X] 我已经在 Github Issues 中搜索过了，但没有找到类似的问题。

### 受影响的包和版本

dev

### 复现步骤

1、冻结后向下滑动，然后点击冻结线上面任意区域
2、将冻结线拖拽到最顶部后滑动表格

### 预期行为

1、点击冻结线上方任意单元格，冻结线下面区域不对自动滚动
2、表格正常显示

### 实际行为

1、表格会自动滚动回来
[2023_12_26 15_33_24 video.webm](https://github.com/dream-num/univer/assets/154207126/2ee4c75c-43b5-4fac-92ba-ad6050c65ca2)

[2023_12_26 15_33_24 commands.json](https://github.com/dream-num/univer/files/13769619/2023_12_26.15_33_24.commands.json)

### 运行环境

_No response_

### 操作系统

_No response_

### 构建工具

_No response_

### LINKED ISSUE #898: [Bug] 冻结行列后，使用滚动条无法拖动表格
### 初始清单

- [X] 这真的是个问题吗？
- [X] 我已经在 Github Issues 中搜索过了，但没有找到类似的问题。

### 受影响的包和版本

dev

### 复现步骤

1、冻结行或冻结列后，使用滚动条滑动查看右侧区域

### 预期行为

可以使用滚动条滑动

### 实际行为

无法滑动
[2023_12_26 20_00_50 commands.json](https://github.com/dream-num/univer/files/13771223/2023_12_26.20_00_50.commands.json)
[2023_12_26 20_00_50 video.webm](https://github.com/dream-num/univer/assets/154207126/80087792-01ec-4463-bf9b-2c222ef0b60a)

### 运行环境

_No response_

### 操作系统

_No response_

### 构建工具

_No response_

### LINKED ISSUE #933: [Bug] 冻结区域显示不对
### 初始清单

- [X] 这真的是个问题吗？
- [X] 我已经在 Github Issues 中搜索过了，但没有找到类似的问题。

### 受影响的包和版本

dev

### 复现步骤

case1、冻结后向下滑动，拖拽冻结行列线
case2、ａ、冻结行后向下话滑动，然后将冻结线拖拽到顶部
　　　ｂ、首列显示不全时，冻结到B列，然后左右滑动

### 预期行为

１、拖拽冻结行列线时行号连续显示，表格自动滚回
image
２、冻结线左侧显示AB列，滑动时表格显示正常

### 实际行为

１、未自动滚回
２、只显示B列被冻结，滑动时表格显示异常
[2023_12_28 17_45_08 commands.json](https://github.com/dream-num/univer/files/13785324/2023_12_28.17_45_08.commands.json)
[2023_12_28 17_45_08 video.webm](https://github.com/dream-num/univer/assets/154207126/322ee817-8278-4902-8c1f-c3a7d3d51268)

### 运行环境

_No response_

### 操作系统

_No response_

### 构建工具

_No response_

### PR DESCRIPTION
**Pull request by weird94**
- [x] I am sure that the code is update-to-date with the `dev` branch.

close #890 
close #898
close #933

### CODE REVIEW

File: packages/sheets-ui/src/controllers/freeze.controller.ts [hunk 1]
```diff
@@ -99,6 +99,8 @@ export class FreezeController extends Disposable {
 
     private _changeToOffsetY: number = 0;
 
+    private _activeViewport: Viewport | null = null;
```
Comment (Jocs) [inline packages/sheets-ui/src/controllers/freeze.controller.ts:102]: Prefer `Nullable<Viewport>`

File: packages/sheets-ui/src/controllers/freeze.controller.ts [hunk 2]
```diff
@@ -387,15 +405,23 @@ export class FreezeController extends Disposable {
...
         this._moveObserver = scene.onPointerMoveObserver.add((moveEvt: IPointerEvent | IMouseEvent) => {
+            const activeViewport = this._getActiveViewport(moveEvt);
+
             const { startX, startY, row, column } = getCoordByOffset(
                 moveEvt.offsetX,
                 moveEvt.offsetY,
                 scene,
                 skeleton,
-                viewMain
+                activeViewport || undefined
```
Comment (Jocs) [inline packages/sheets-ui/src/controllers/freeze.controller.ts:424]: `_getActiveViewport` 没有 activeViewport 直接返回 undefined，这儿就不用单独处理呢？

Comment (weird94) [reply packages/sheets-ui/src/controllers/freeze.controller.ts:426]: 是为了解决类型问题 _getActiveViewport可能返回null

**Review by wzhudev — CHANGES_REQUESTED** [review]
Should have unit test to cover changes.

**Review by wzhudev — APPROVED** [review]

**Pull request merged**